compose/mdns.yml

```diff
@@ -0,0 +1,31 @@
+version: "2.0"
+
+services:
+  balena-mdns-publisher:
+    image: balena/balena-mdns-publisher:master
+    network_mode: "host"
+    cap_add:
+        - SYS_RESOURCE
+        - SYS_ADMIN
+    security_opt:
+        - apparmor:unconfined
+    tmpfs:
+        - /run
+        - /sys/fs/cgroup
+    # balenaOS - Required for host DBus comms. Not required for standalone Linux
+    labels:
+        io.balena.features.dbus: '1'
+        io.balena.features.supervisor-api: '1'
+    environment:
+        CONFD_BACKEND: ENV
+        # The name of the TLD to use. This *must* match certificates used for the rest of
+        # the resin backend (eg. that for BALENA_ROOT_CA if present).
+        MDNS_TLD: ${OPENBALENA_HOST_NAME}
+        # List of subdomains to advertise. This must include all required hosts.
+        MDNS_SUBDOMAINS: '["api", "db", "registry", "s3", "vpn"]'
+        # The expectation is the DBus socket to use is always at the following location.
+        DBUS_SESSION_BUS_ADDRESS: "unix:path=/host/run/dbus/system_bus_socket"
+        # Selects the interface used for incoming connections from the wider subnet.
+        # For NUCs, this is `eno1`. If running natively, pick the appropriate interface.
+        # Alternatively, keep the default commented out to autoselect.
+        #INTERFACE: "eno1"
```

Comment:
It is massively worth documenting this in full in the OB docs, else it's potentially going to confuse people if they think it'll autoselect an interface under non-balena Linux.

Reply:
That's fine. As a question though, why can't it auto-select the interface in non-balenaOS instances? Could we make it so that this is possible?

Reply:
@richbayliss Oh you absolutely could... to a degree. But the question is, when you find three separate IPv4 interfaces, which one do you use? The MDNS publisher was originally going to be used purely in BoB, where usually someone has a single internet facing interface (or if both are connected, it doesn't matter), where the Supervisor does a load of filtering to ensure that none of the interfaces it knows won't be internet connected (or a valid route) are scrubbed. How are you going to do this under Linux? :)